osu.Game/Screens/Select/Carousel/CarouselBeatmapSet.cs

```diff
@@ -4,30 +4,56 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using osu.Framework.Bindables;
 using osu.Framework.Extensions.IEnumerableExtensions;
 using osu.Game.Beatmaps;
 using osu.Game.Screens.Select.Filter;
+using osu.Game.Users;
 
 namespace osu.Game.Screens.Select.Carousel
 {
     public class CarouselBeatmapSet : CarouselGroupEagerSelect
     {
+        private readonly Bindable<User> localUser;
+
         public IEnumerable<CarouselBeatmap> Beatmaps => InternalChildren.OfType<CarouselBeatmap>();
 
         public BeatmapSetInfo BeatmapSet;
 
-        public CarouselBeatmapSet(BeatmapSetInfo beatmapSet)
+        public CarouselBeatmapSet(BeatmapSetInfo beatmapSet, Bindable<User> localUser)
         {
             BeatmapSet = beatmapSet ?? throw new ArgumentNullException(nameof(beatmapSet));
 
             beatmapSet.Beatmaps
                       .Where(b => !b.Hidden)
                       .Select(b => new CarouselBeatmap(b))
                       .ForEach(AddChild);
+
+            this.localUser = localUser;
         }
 
         protected override DrawableCarouselItem CreateDrawableRepresentation() => new DrawableCarouselBeatmapSet(this);
 
+        protected override CarouselItem GetNextToSelect()
+        {
+            if (LastSelected == null)
+            {
+                decimal? pp = localUser.Value?.Statistics?.PP ?? 60; // TODO: This needs to get ruleset specific statistics
+
+                var recommendedDifficulty = Math.Pow((double)pp, 0.4) * 0.195;
```

Comment:
Not a fan of this being duplicated lazer- and web-side. Candidate for inclusion in API, perhaps?

Reply:
I think it's fine to have this in both places, with a comment linking them. We don't really want to re-retrieve recommended difficulties every time we update this algorithm.